crates/pathfinder/src/sequencer.rs

```diff
@@ -1038,11 +1060,78 @@ mod tests {
                 (StatusCode::TOO_MANY_REQUESTS, ""),
             ]);
 
-            let error = run_retry(statuses).await.unwrap_err();
+            let (_jh, addr) = status_queue_server(statuses);
+            let error = super::retry(|| async {
+                let mut url = reqwest::Url::parse("http://localhost/").unwrap();
+                url.set_port(Some(addr.port())).unwrap();
+                let resp = reqwest::get(url).await?;
+                super::parse::<String>(resp).await
+            })
+            .await
+            .unwrap_err();
             assert_matches!(
                 error,
                 SequencerError::TransportError(te) => assert_eq!(te.status(), Some(StatusCode::SERVICE_UNAVAILABLE))
             );
         }
+
+        #[tokio::test]
+        #[traced_test]
+        async fn client_timeout() {
```

Comment:
Aren't these two test basically the same?

I mean the only difference seems to be is that for `client_timeout()` the request timeout is set via the `ClientBuilder` and for `request_timeout()` via `RequestBuilder`. The result should be effectively the same, so I don't quite see the point in having two separate test cases.

Reply:
Indeed, it's the server that determines the type of timeout here, regardless of whichever client api is used. I'll make it a single test case :+1: